### PR TITLE
Separate cycler for each model

### PR DESCRIPTION
**Pull request by merifri**

### CODE REVIEW

File: storage/cycler.go [hunk 1]
```diff
@@ -44,10 +44,12 @@ func (c *Cycler) shiftByKeep(keep int) (first *Package) {
 	return
 }
 
-func (c *Cycler) run(fileKey string, keep int, deletePackage func(fileKey string) error) {
-	c.load()
+func (c *Cycler) run(model string, fileKey string, keep int, deletePackage func(fileKey string) error) {
+	cyclerFileName := path.Join(cyclerPath, model + ".cycler.json")
```
Comment (huacnlee) [inline storage/cycler.go]:
```go
path.Join(cyclerPath, "cycler", model + ".json")
```

So, the model cycler filename will be: `.gobackup/cycler/.foo.json` not `.gobackup/foo.cycler.json`

File: storage/cycler.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ type Package struct {
 }
 
 var (
-	cyclerFileName = path.Join(config.HomeDir, ".gobackup", "cycler.json")
+	cyclerPath = path.Join(config.HomeDir, ".gobackup")
```
Comment (huacnlee) [inline storage/cycler.go]:
```go
cyclerPath = path.Join(config.HomeDir, ".gobackup/cycler")
```

**Comment by huacnlee** [comment]
Can you tell me, why do you need to separate it?

**Comment by merifri** [comment]
Config:

```yaml
models:
    daily:
        compress_with:
            type: tgz
        databases:
            app1:
                database: app
                type: postgresql
        store_with:
            keep: 10
            path: /mnt/backups/postgres
            type: local
    weekly:
        compress_with:
            type: tgz
        databases:
            app1:
                database: app
                type: postgresql
        store_with:
            access_key_id: ...
            bucket: ...
            keep: 7
            path: ''
            region: eu-central-1
            secret_access_key: ...
            type: s3
```

```console
#Ansible: Backup job: daily
15 5 * * * /usr/local/bin/gobackup perform --model daily
#Ansible: Backup job: weekly
30 5 * * 1 /usr/local/bin/gobackup perform --model weekly
```

Each model will remove the unknown entries from the file `cycle.json`. Old backups will not be removed from storages.

File: storage/cycler.go [hunk 3]
```diff
@@ -66,7 +68,11 @@ func (c *Cycler) run(fileKey string, keep int, deletePackage func(fileKey string
 	}
 }
 
-func (c *Cycler) load() {
+func (c *Cycler) load(cyclerFileName string) {
+	if !helper.IsExistsPath(cyclerPath) {
+		helper.Exec("mkdir", "-p", cyclerPath)
```
Comment (huacnlee) [inline storage/cycler.go]: helper.MkdirP(cyclerPath)

and you must rebase master that has fixed CI.

Comment (merifri) [reply storage/cycler.go]: Thanks, fixed.

**Pull request merged**